src/components/form/text-input.tsx

```diff
@@ -0,0 +1,35 @@
+import cx from "classnames";
+import { HTMLProps, forwardRef } from "react";
+import { UseFormRegisterReturn } from "react-hook-form";
+
```

Comment:
`UseFormRegisterReturn<string>` - These are the props that come from the register function return type for a string
`HTMLProps<HTMLInputElement>` - These are the props that are inherent for <input> tags